app/src/main/java/org/listenbrainz/android/repository/user/UserRepositoryImpl.kt

```diff
@@ -0,0 +1,43 @@
+package org.listenbrainz.android.repository.user
+
+import org.listenbrainz.android.model.Listens
+import org.listenbrainz.android.model.PinnedRecording
+import org.listenbrainz.android.model.ResponseError
+import org.listenbrainz.android.model.user.TopArtists
+import org.listenbrainz.android.model.user.UserSimilarityPayload
+import org.listenbrainz.android.service.UserService
+import org.listenbrainz.android.util.Resource
+import org.listenbrainz.android.util.Utils.parseResponse
+import javax.inject.Inject
+
+class UserRepositoryImpl @Inject constructor(
+    private val service: UserService
+) : UserRepository {
+    override suspend fun fetchUserListenCount(username: String?): Resource<Listens?> = parseResponse {
+        if(username.isNullOrEmpty()) return ResponseError.DOES_NOT_EXIST.asResource()
+        service.getListenCount(username)
+    }
+
+    override suspend fun fetchListeningNow(username: String?): Resource<Listens?> {
+        TODO("Not yet implemented")
```

Comment:
This is already present right? Will you be moving that from Socket Repo to here?

Reply:
Apparently, this was not working as of date as Jasjeet had pointed out.. so was looking to reimplement it , now that I realize it was already present, I'll just make reqd changes in the original one